extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -96,13 +125,64 @@ export class NetCoreTool {
 		return undefined;
 	}
 
+	/**
+	 * This function checks if the installed dotnet version is 3.1 or greater.
+	 * Versions lower than 3.1 aren't supported for building projects.
+	 * Returns: True if installed dotnet version is supported, false otherwise.
+	 * 			Undefined if dotnet isn't installed.
+	 */
+	private async isNetCoreVersionSupported(): Promise<boolean | undefined> {
+		try {
+			const spawn = child_process.spawn;
+			let child: child_process.ChildProcessWithoutNullStreams;
+			let isSupported: boolean | undefined = undefined;
+
+			child = spawn('dotnet --version', [], {
+				shell: true
+			});
+
+			child.stdout.on('data', (data) => {
+				this.netCoreSdkInstalledVersion = String(data);
+			});
+
+			isSupported = await this.onExit(child);
+			if (!isSupported) {
+				this.netCoreInstallState = netCoreInstallState.netCoreVersionNotSupported;
+			}
+			return isSupported;
+		} catch (err) {
+			console.log(err);
+			return undefined;
+		}
+	}
+
+	private onExit(childProcess: child_process.ChildProcess): Promise<boolean | undefined> {
```

Comment:
This isn't really clear what it's doing or what the return value means. Would suggest updating the name and/or adding documentation to make this clear.

Reply:
Removed this separate method and merged it in the parent method.